lib/commands/network.js

```diff
@@ -0,0 +1,50 @@
+import log from '../logger';
+
+let commands = {}, helpers = {}, extensions = {};
+
+commands.getNetworkConnection = async function () {
+  log.info('Getting network connection');
+  let airplaneModeOn = await this.adb.isAirplaneModeOn();
+  let connection = airplaneModeOn ? 1 : 0;
+
+  // no need to check anything else if we are in airplane mode
+  if (!airplaneModeOn) {
+    let wifiOn = await this.adb.isWifiOn();
+    connection += (wifiOn ? 2 : 0);
+    let dataOn = await this.adb.isDataOn();
+    connection += (dataOn ? 4 : 0);
+  }
+
+  return connection;
+};
+
+commands.setNetworkConnection = async function (type) {
+  log.info('Setting network connection');
+  // decode the input
+  let airplaneMode = type % 2;
+  type >>= 1;
+  let wifi = type % 2;
+  type >>= 1;
+  let data = type % 2;
+
+  await this.handleADBDisconnectAndPerformAction('setAirplaneMode', airplaneMode);
+  await this.handleADBDisconnectAndPerformAction('broadcastAirplaneMode', airplaneMode);
+
+  if (!airplaneMode){
+    await this.handleADBDisconnectAndPerformAction('setWifiAndData', {wifi, data});
+  }
+
+  return await this.getNetworkConnection();
+};
+
+helpers.handleADBDisconnectAndPerformAction = async function(action, params){
+  this.ignoreUnexpectedShutdown = true;
+  await this.adb[action](params);
+  await this.adb.restart();
+  await this.bootstrap.start(this.opts.appPackage, this.opts.disableAndroidWatchers);
```

Comment:
the fact that the uiautomator bootstrap shuts down makes me think that this function is really `handleUiAutomatorDisconnectAndPerformAction`. The long name also makes me wonder if we should be architecting this differently. What if it was:

```js
// how we call it
await this.wrapBootstrapDisconnect(async () => {
  await this.adb.setAirplaneMode(airplaneMode);
});

// how it's defined
helpers.wrapBootstrapDisconnect = async function (wrapped) {
  this.ignoreUnexpectedShutdown = true;
  await wrapped();
  await this.adb.restart();
  await this.bootstrap.start(this.opts.appPackage, this.opts.disableAndroidWatchers);
};
```

This seems a lot cleaner and has the advantage of having clearer names :-)